### PR TITLE
cantoolz: init at 3.7.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
CANToolz is a framework for analysing CAN networks and devices. It
provides multiple modules that can be chained using CANToolz's pipe
system and used by security researchers, automotive/OEM security
testers in black-box analysis.

CANToolz can be used for ECU discovery, MitM testing, fuzzing, brute 
forcing, scanning or R&D, testing and validation. More can easily be
implemented with a new module.

https://github.com/CANToolz/CANToolz

Related to #81418

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/cantoolz/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    owner = "CANToolz";
+    repo = "CANToolz";
+    rev = "v${version}";
+    sha256 = "0xkj7zyx6pz866q61c84mdagpgdyd633v85hk7qxhamca33rc4yi";
+  };
+
+  patches = [
+    (fetchpatch {
+      # https://github.com/CANToolz/CANToolz/pull/30
+      url = "https://github.com/CANToolz/CANToolz/pull/30.patch";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/cantoolz/default.nix:27]: Please fetch a commit and add a comment why this is needed.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108701` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>cantoolz</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 108701` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>cantoolz</li>
  </ul>
</details>

File: pkgs/tools/networking/cantoolz/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/tools/networking/cantoolz/default.nix]: Updated

Thanks for the feedback.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 108701` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cantoolz</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 108701` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>cantoolz</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108701` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>cantoolz</li>
</ul>
</details>

**Pull request merged**